code/game/machinery/computer/bhangmeter.dm

```diff
@@ -0,0 +1,335 @@
+var/list/bhangmeters = list()
+var/list/sensed_explosions = list()
+
+/*
+	/obj/machinery/computer/bhangmeter
+	/datum/station_holomap/bhang
+	/datum/sensed_explosion
+*/
+
+/obj/machinery/computer/bhangmeter
+	name = "bhangmeter"
+	desc = "Uses a tachyon-doppler array to measure explosions of all shapes and sizes, as well as detecting meteors on a collision course with the station."
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "bhangmeter"
+	circuit = "/obj/item/weapon/circuitboard/bhangmeter"
+
+	light_color = "#FFD400"
+
+	var/mob/watching_mob = null
+	var/list/watching_mobs = list()
+	var/list/watcher_maps = list()
+	var/list/watcher_buttons = list()
+
+	var/original_zLevel = 1	//zLevel on which the machine was initialized.
+	var/forced_zLevel = 0	//can be set by mappers to override the Station Map's zLevel
+							//only zLevels to have an actual map displayed are the Station, the Asteroid, and the Derelict.
+
+	var/datum/station_holomap/bhang/holomap_datum
+
+	var/announcement_cooldown = 5 SECONDS
+	var/last_announcement = 0
+	var/datum/sensed_explosion/last_announced_explosion
+
+/obj/machinery/computer/bhangmeter/New()
+	..()
+	if (forced_zLevel)
+		original_zLevel = forced_zLevel
+	else
+		original_zLevel = loc.z
+	bhangmeters += src
+	if(ticker && holomaps_initialized)
+		initialize()
+
+/obj/machinery/computer/bhangmeter/Destroy()
+	bhangmeters -= src
+	stopWatching()
+	holomap_datum = null
+	..()
+
+/obj/machinery/computer/bhangmeter/initialize()
+	var/turf/T
+	if (forced_zLevel)
+		var/turf/U = get_turf(src)
+		T = locate(U.x,U.y,forced_zLevel)
+		original_zLevel = forced_zLevel
+	else
+		T = get_turf(src)
+		original_zLevel = T.z
+
+	holomap_datum = new()
+	holomap_datum.initialize_holomap(T)
+
+/obj/machinery/computer/bhangmeter/process()
+	. = ..()
+	if (.)
+		for (var/mob/M in watching_mobs)
+			if (M.client)
+				M.client.images -= watcher_maps["\ref[M]"]
+				qdel(watcher_maps["\ref[M]"])
+				watcher_maps["\ref[M]"] = image(holomap_datum.get_bhangmap())
+				var/image/I = watcher_maps["\ref[M]"]
+				I.loc = M.hud_used.holomap_obj
+				M.client.images |= watcher_maps["\ref[M]"]
+	else
+		stopWatching()
+
+/obj/machinery/computer/bhangmeter/say_quote(text)
+	return "coldly states, [text]"
+
+/obj/machinery/computer/bhangmeter/attack_hand(var/mob/user)
+	if(isliving(user) && anchored && !(stat & (FORCEDISABLE|NOPOWER|BROKEN)))
```

Comment:
attack_ghost points here but this sanity check kicks out observers, but it also doesn't care if you are dead or a mouse

you can also invert the check with a return to avoid rightward creep